src/main/java/Parser/DateTimeParser.java

```diff
@@ -65,15 +59,9 @@
         String weekDate = "";
         dateStringSplit = dateTimeStringSplit[0].trim().split(" ");
         weekDate = dateStringSplit[0];
-        if (weekDate.equalsIgnoreCase("reading") || weekDate.equalsIgnoreCase("exam")
-                || weekDate.equalsIgnoreCase("week") || weekDate.equalsIgnoreCase("recess")){
-            weekDate = input.substring(0,input.length()- 4); // week x day y
-            String time = input.substring(input.length()- 4).trim(); // time E.g 0300
-            weekDate = LT.getValue(weekDate) + " " + time;
-        } else {
-            String time = input.substring(input.length()- 4).trim();
-            weekDate = dateTimeStringSplit[0] + " " + time;
-        }
+        weekDate = WeekFormatParse.acadWeekToString(weekDate,input.substring(0,input.length()-4));
+        String time = input.substring(input.length()- 4).trim();
```

Comment:
If you're aware of the input it might be better to use '.replacefirst 'method instead of .'substring(x**-4**)' might encounter arrayoutofbound situation.